### DISCUSSION TITLE
is there any extension for webstorm?

### DISCUSSION THREAD

**Discussion by inside5545** [oxc-project/oxc#3269] [Feature Request]
is there any extension for webstorm?

**Comment by IWANABETHATGUY** [comment]
If you are using the paid JetBrain products, the integration should be pretty simple, https://blog.jetbrains.com/platform/2023/07/lsp-for-plugin-developers/

**Comment by bc-m** [comment]
Maybe it is worth taking a look at [Biomes IntelliJ plugin](https://github.com/biomejs/biome-intellij).

**Comment by IWANABETHATGUY** [comment]
I am already working on the intellij plugin

**Comment by IWANABETHATGUY** [comment]
A MVP version has been finished, https://github.com/IWANABETHATGUY/oxc-intellij-plugin/tree/main, 
I will publish it to market ASAP

**Comment by Boshen** [comment]
> A MVP version has been finished, https://github.com/IWANABETHATGUY/oxc-intellij-plugin/tree/main, I will publish it to market ASAP

Can you move it to the oxc-project org?

**Comment by IWANABETHATGUY** [comment]
> > A MVP version has been finished, https://github.com/IWANABETHATGUY/oxc-intellij-plugin/tree/main, I will publish it to market ASAP
> 
> Can you move it to the oxc-project org?

I would move to `oxc-project` after I clean up the code.

**Comment by winjeysong** [comment]
> > > A MVP version has been finished, https://github.com/IWANABETHATGUY/oxc-intellij-plugin/tree/main, I will publish it to market ASAP
> > 
> > 
> > Can you move it to the oxc-project org?
> 
> I would move to `oxc-project` after I clean up the code.

Thanks for your working!
When will the plugin release. I'm really looking forward to that! 😊

**Comment by bc-m** [comment]
Would love to use oxc in our projects, but we use webstorm instead of vscode. Losing inline lints feels like a downgrade yet. Is this issue open for help or are there any blockers? 

If I can help you with testing feel free to ping me. Thank you for your work!

**Comment by adoin** [comment]
It seems no plugin release for now?

**Comment by Smrtnyk** [comment]
Any progress on this?
I am really looking forward to oxc but lack of plugin for webstorm is pretty huge blocker

**Reply by camc314** [reply]
have you tried https://github.com/IWANABETHATGUY/oxc-intellij-plugin/tree/main ?

**Reply by Smrtnyk** [reply]
ah ok, I thought there was something planned to be released in the plugins store

**Reply by camc314** [reply]
doesn't look like it's there yet i'm afraid.

You can try installing it manuallt in the mean time?

<img width="849" alt="Screenshot 2024-11-11 at 21 25 50" src="https://github.com/user-attachments/assets/2b955c2c-3e2f-4c26-bdfa-5d2257cd68ae">

**Reply by Smrtnyk** [reply]
I will, but it is a bit cumbersome proposing it to my coworkers at my work
I will use it for now in my hobby project for personal use to first get some experience with oxc linter
Thnx for the responses

**Reply by IWANABETHATGUY** [reply]
> Any progress on this? I am really looking forward to oxc but lack of plugin for webstorm is pretty huge blocker

There is one left task https://github.com/oxc-project/oxc-intellij-plugin/issues/84 that may block the original release, but maybe we could release  it first.

**Reply by nrayburn-tech** [reply]
Waiting for nested configuration support was the plan. I’m not sure when that’s expected to be completed though. Finishing out the overrides support was first and that looks to still be in progress https://github.com/oxc-project/oxc/pull/6974.

@DonIsaac might be able to share an update on when it might be done and his thoughts on whether we should just add support for a single config file and release the plugin in the meantime.

**Reply by Smrtnyk** [reply]
I have built the extension from the source
I had an issue that I was using node via nvm on ubuntu and didn't have a node alias for `#!/usr/bin/env node`
but after adding alias oxc plugin wasn't crashing anymore.
Oxc language server was now running and plugin was connected to it.

But the issue I am having now is that my config file `oxlint.json` that I have in the root of my project is not being picked up.
Doesn't matter which rule I set in the config I see no errors in the IDe, it works when running lint with cli command, but intellij plugin is not flagging errors in the file, I don't see errors in my IDE.
Also seems like this plugin is not running on vue files, altough when linting via cli it works perfectly fine.

**Reply by nrayburn-tech** [reply]
I don’t have the code open, but I do believe both of your issues are expected. The code hasn’t been setup for that.

For config files, I was waiting for some additional config changes in Oxlint before adding anything to the plugin. Currently I think it only picks up a single config file name, I don’t know what it is though.
I don’t think anything has changed here, still waiting on some Oxlint config enhancements before doing much in the plugin here. Tracked on the plugin side here https://github.com/oxc-project/oxc-intellij-plugin/issues/84.

For Vue files, I think I only added js,jsx, and the TypeScript equivalent files. So there’s quite a few missing that can be added including Vue.
I think this is just file extensions and should be a pretty easy fix. I’ll try to get to it soon, or somebody can submit a PR for it.

**Reply by Smrtnyk** [reply]
Yeah I added vue to the list of supported extensions and I can see now that oxc is running on vue files now.
I just don't know where to modify extension to make it pick up my oxlint.json file for the config, I am not that familiar with intellij extensions, especially not with oxc stuff.

**Reply by nrayburn-tech** [reply]
The default is in the language server currently, which is part of the oxc_language_server crate in this repo. If you use a “.eslintrc” file it should be picked up.

This can be configured from the plugin as well, but it’s not as easy to explain. I believe it would be done by modifying [this](https://github.com/oxc-project/oxc-intellij-plugin/blob/0446c29788b20645f58c9e14c06546a2b04c41f0/src/main/kotlin/com/github/oxc/project/oxcintellijplugin/lsp/OxcLspServerDescriptor.kt#L38) method to return the correct settings. The format in JSON is something like
{“settings”: {“configPath”: “your/path/oxlint.json”}}

**Reply by Smrtnyk** [reply]
tnx a lot for the pointers @nrayburn-tech, I really appreciate it
I have managed to make it work by hacking the method you mentioned
by doing something like this
```kotlin
override fun createInitializeParams(): InitializeParams {
        val params = super.createInitializeParams()

        val configPath = findConfigPath()
        if (configPath != null) {
            params.initializationOptions = mapOf(
                "settings" to mapOf(
                    "enable" to true,
                    "run" to "onType",
                    "configPath" to configPath  
                )
            )
            thisLogger().debug("Added initializationOptions with configPath: $configPath")
        } else {
            thisLogger().warn("oxlint.json configuration file not found.")
        }

        return params
    }
 ```
 `findConfigPath()` is a helper I added to find the file
 I had to specify all 3 params because it seems that rust crate didn't consider the options if I only provided the configPath. 
 I am able to see lint issues now both in ts and vue files

**Comment by nrayburn-tech** [comment]
There is a plugin available for Oxc now. GitHub page is https://github.com/oxc-project/oxc-intellij-plugin which contains information on how to find it on the marketplace or install manually.
